kuwala/core/canvas/src/data/state/models/CanvasModel.js

```diff
@@ -165,6 +196,10 @@ const CanvasModel = {
                 targetDTO = targetElement.data.transformationBlock;
                 targetDTO.connectedSourceNodeIds = targetDTO.connectedSourceNodeIds.filter((el) => el !== source);
                 updateTransformationBlock({ addNode, elements, setElements, updatedBlock: targetDTO });
+            } else if (targetElement.type === EXPORT_BLOCK) {
+                targetDTO = targetElement.data.exportBlock;
+                targetDTO.connectedSourceNodeIds = targetDTO.connectedSourceNodeIds.filter((el) => el !== source);
+                updateExportBlock({ addNode, elements, setElements, updatedBlock: targetDTO });
```

Comment:
This action can be moved outside of all the blocks and only conditionally set the `targetDTO`.